html/semantics/scripting-1/the-script-element/moving-between-documents.html

```diff
@@ -0,0 +1,139 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Moving script elements between documents</title>
+<link rel="author" href="mailto:d@domenic.me" title="Domenic Denicola">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/#execute-the-script-block">
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<!-- Background:
+
+- https://www.w3.org/Bugs/Public/show_bug.cgi?id=11323
+- https://github.com/whatwg/html/issues/2137
+- https://github.com/whatwg/html/issues/2469
+- https://github.com/whatwg/html/pull/2673
+
+-->
+
+<body>
+<script>
+"use strict";
+
+for (const type of ["text/javascript", "module"]) {
+  async_test(t => {
+    t.add_cleanup(() => {
+      window.didExecute = undefined;
+    });
+
+    const iframe = document.createElement("iframe");
+    iframe.onload = t.step_func_done(() => {
+      iframe.contentDocument.write("<streaming-element>");
+      document.body.appendChild(iframe.contentDocument.querySelector("streaming-element"));
+      iframe.contentDocument.write(`<script id="s1" type="${type}">window.didExecute = true;<` + "/script>");
+      iframe.contentDocument.write("</streaming-element>");
+
+      const s = document.querySelector("#s1");
+      s.onload = t.unreached_func("onload");
+      s.onerror = t.unreached_func("onerror");
+
+      iframe.contentDocument.close();
+
+      assert_equals(window.didExecute, undefined, "The script must not have executed in this window");
+      assert_equals(iframe.contentWindow.didExecute, undefined,
+        "The script must not have executed in the iframe window");
+    });
+
+    document.body.appendChild(iframe);
+  }, `${type}: moving to another document during parsing, inline script`);
+
+  async_test(t => {
+    t.add_cleanup(() => {
+      window.didExecute = undefined;
+    });
+
+    const iframe = document.createElement("iframe");
+    iframe.onload = t.step_func(() => {
+      iframe.contentDocument.write("<streaming-element>");
+      document.body.appendChild(iframe.contentDocument.querySelector("streaming-element"));
+      iframe.contentDocument.write(`<script id="s2" type="${type}" src="resources/flag-setter.js"><` + "/script>");
+      iframe.contentDocument.write("</streaming-element>");
+
+      const s = document.querySelector("#s2");
+      s.onload = t.unreached_func("onload");
+      s.onerror = t.unreached_func("onerror");
+
+      iframe.contentDocument.close();
+
+      t.step_timeout(() => {
+        assert_equals(window.didExecute, undefined, "The script must not have executed in this window");
+        assert_equals(iframe.contentWindow.didExecute, undefined,
+          "The script must not have executed in the iframe window");
+        t.done();
+      }, 3000);
```

Comment:
Why a timeout here? Since the parser inserts the script element, wouldn't it simply block on it loading if a user agent doesn't implement the spec?

Reply:
That's also why the belated onload/onerror handlers seem problematic. They probably need to be in the markup, no?

Reply:
They should probably be before setting s.src, it's true.